test/create-snowpack-app/create-snowpack-app.test.js

```diff
@@ -0,0 +1,96 @@
+const fs = require('fs');
+const path = require('path');
+const execa = require('execa');
+const dircompare = require('dir-compare');
+
+const TEMPLATES_DIR = path.resolve(__dirname, '..', '..', 'packages', '@snowpack');
+
+const templates = fs.readdirSync(TEMPLATES_DIR).filter((dir) => dir.startsWith('app-template-'));
+
+const format = (stdout) =>
+  stdout
+    .replace(/([\w\-]+\-)[a-z0-9]{8}(\.js)/g, '$1XXXXXXXX$2') // strip chunk hash
+    .replace(/((\s+$)|((\\r\\n)|(\\n)))/gm, '') // strip whitespace chars
+    .replace(/\n\s*\/\*[^*]+\*\/\s*\n/gm, '\n'); // strip full-line comments (throws Svelte test)
+
+describe('create-snowpack-app', () => {
+  // test npx create-snowpack-app bin
+  it('npx create-snowpack-app', async () => {
+    const template = 'app-template-preact'; // any template will do
+
+    // run the local create-snowpack-app bin
+    await execa(
+      'node',
+      [
+        './packages/create-snowpack-app',
+        `./test/create-snowpack-app/test-install`,
+        '--template',
+        `../../../packages/@snowpack/${template}`,
+        '--use-yarn', // we use Yarn for this repo
+        '--force', // saves you from having to manually delete things
+      ],
+      {cwd: path.resolve(__dirname, '..', '..')},
+    );
+
+    // snowpack.config.json is a file we can test for to assume successful
+    // install, since it’s added at the end.
+    const snowpackConfigExists = fs.existsSync(
+      path.resolve(__dirname, 'test-install', 'snowpack.config.json'),
+    );
+    expect(snowpackConfigExists).toBe(true);
+  });
+
+  // template snapshots
+  templates.forEach((template) => {
+    it(`--template @snowpack/${template}`, async () => {
+      const cwd = path.join(TEMPLATES_DIR, template);
+
+      // 1. build
+
+      // pre-build: set config to minify output
```

Comment:
This is the only annoying part of this test. We want to test our templates against a snapshot to ensure nothing intentionally changed. However, we need _unminified_ snapshots. And we need to get that without the tests affecting the user config.

So the best I could come up with is to modify the config in each template to disable minification, then restore the original config after the test runs. Suggestions welcome on a better solution here.